### PR TITLE
✨ Add today variable with current date

### PR DESCRIPTION
**Pull request by arassaria**
# Description

Add today variable with current date to be able to get rides from datebase for current date. 😄 🚀 

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: client/src/pages/Rides.js [hunk 1]
```diff
@@ -7,6 +7,9 @@ import Card from "../components/Card";
 import HeaderMain from "../components/HeaderMain";
 import ButtonPlus from "../components/ButtonPlus";
 import { useHistory } from "react-router-dom";
+import { getCurrentDate } from "../utils/date";
+import { getDataByQuery } from "../utils/api";
+import { useEffect } from "react";
```
Comment (phlgr) [inline client/src/pages/Rides.js]: Sort your imports according to importance e.g. React at the top. Feel free to ignore this as this is purely code style preference

File: client/src/pages/Rides.js [hunk 2]
```diff
@@ -24,6 +27,19 @@ const PageWrapper = styled.div`
...
   const history = useHistory();
+  useEffect(() => {
+    const doFetch = async () => {
+      const today = getCurrentDate();
+      const todaysRides = await getDataByQuery({
+        collectionName: "tasks",
+        dataName: "date",
+        query: today,
+      });
+      console.log(todaysRides);
```
Comment (phlgr) [inline client/src/pages/Rides.js]:
```suggestion

```

File: client/src/utils/date.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+export function getCurrentDate() {
+  const currentDate = new Date();
+  // Following code is super hacky, gonna be changed, just needed it to work for now.
+  // Need fixed format for currentDate to be able to read data from database.
```
Comment (phlgr) [inline client/src/utils/date.js]: yes.

**Review by phlgr — CHANGES_REQUESTED** [review]

File: client/src/pages/Rides.js (hunk 1 above)
Comment (arassaria) [reply client/src/pages/Rides.js]: Sorted them, thanks for the advice 😄 🚀

File: client/src/pages/Rides.js (hunk 2 above)
Comment (arassaria) [reply client/src/pages/Rides.js]: Forgot to remove it. Thanks for reminding 🚀

File: client/src/utils/date.js (hunk 3 above)
Comment (arassaria) [reply client/src/utils/date.js]: @phlgr Any sort of advice on how to simplify this part of the code to still always get the data formed as "yyyy-mm-dd"?

File: client/src/utils/date.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+    currentDate.getMonth() + 1 < 10
+      ? `0${currentDate.getMonth() + 1}`
+      : currentDate.getMonth() + 1
+  }-${
+    currentDate.getDate() < 10
+      ? `0${currentDate.getDate()}`
+      : currentDate.getDate()
+  }`;
+  return today;
+}
```
Comment (lmachens) [inline client/src/utils/date.js]:
```suggestion
function pad(number) {
  return (number < 10) ? `0${number}` : number;
}

export function getCurrentDateString() {
  const date = new Date();
  return `${date.getFullYear()}-${pad(date.getMonth())}-${pad(date.getDay())}`;
}
```

or (more magic)

```suggestion
export function getCurrentDateString() {
  return new Date().toISOString().substr(0, 10);
}
```

File: client/src/pages/Rides.js [hunk 5]
```diff
@@ -24,6 +29,18 @@ const PageWrapper = styled.div`
...
   const history = useHistory();
+  useEffect(() => {
+    const doFetch = async () => {
+      const today = getCurrentDate();
+      const todaysRides = await getDataByQuery({
+        collectionName: "tasks",
+        dataName: "date",
+        query: today,
+      });
+      return todaysRides;
```
Comment (lmachens) [inline client/src/pages/Rides.js]: `todaysRides` is not used because `doFetch` doesn't expect a return value.
Maybe you like to save it in a state?

**Review by lmachens — COMMENTED** [review]
gj

File: client/src/utils/date.js (hunk 4 above)
Comment (arassaria) [reply client/src/utils/date.js]: Thanks, works fine 😄

File: client/src/pages/Rides.js (hunk 5 above)
Comment (arassaria) [reply client/src/pages/Rides.js]: Changed it to a state. Thanks for the advice. 🚀

File: client/src/utils/date.js (hunk 3 above)
Comment (arassaria) [reply client/src/utils/date.js]: Nvm, has been solved.

**Review by bwnstck — APPROVED** [review]
lgtm 👩‍🎤

**Pull request merged**